src/charm.py

```diff
@@ -561,6 +578,16 @@ def _on_anonymize_user_action(self, event: ActionEvent) -> None:
             }
         )
 
+    def _on_oauth_relation_changed(self, _: RelationChangedEvent) -> None:
+        """Handle oauth_relation_changed event."""
+        charm_state = self.build_charm_state()
+        mas_configuration = MASConfiguration.from_charm(self)
+        oauth_client_config = generate_oauth_client_config(
+            mas_configuration, charm_state.synapse_config
+        )
+        self._oauth.update_client_config(oauth_client_config)
```

Comment:
Wouldnt this trigger another relation-changed?

Reply:
I put it here because the oauth client config that we generate is mostly constant, in which case it won't fire a relation changed event.

The only part that can change is the `redirect_uri` which changes if we update the `public_baseurl` config, in which case we'd need to fire a relation changed event again so that the provider updates its oauth client on that side.